athena/llm_core/llm_core/loaders/model_loaders/azure_loader.py

```diff
@@ -0,0 +1,60 @@
+from enum import Enum
+import os
+import requests
+from typing import Dict, List
+
+from langchain.base_language import BaseLanguageModel
+from langchain_openai import AzureChatOpenAI
+
+from athena.logger import logger
+
+
+# Discover available models from Azure 
+AZURE_OPENAI_PREFIX = "azure_openai_"
+azure_available = bool(os.environ.get("AZURE_OPENAI_API_KEY"))
+
+azure_available_models: Dict[str, BaseLanguageModel] = {}
+
+# Azure OpenAI
+if azure_available:
+    def _get_azure_openai_deployments() -> List[str]:
+        base_url = f"{os.environ.get('AZURE_OPENAI_ENDPOINT')}/openai"
+        headers = {"api-key": os.environ["AZURE_OPENAI_API_KEY"]}
+
+        models_response = requests.get(
+            f"{base_url}/models?api-version=2023-03-15-preview",
+            headers=headers, timeout=30
+        )
```

Comment:
Not sure if needed - but I guess we can store the API version as a constant or get it from a config/yml